tests/test_raster/test_interpolate.py

```diff
@@ -0,0 +1,568 @@
+
+        for i in range(len(points_x_in)):
+            xlow = int(index_x_in[i] - 0.5)
+            xupp = int(index_x_in[i] + 0.5)
+            ylow = int(index_y_in[i] - 0.5)
+            yupp = int(index_y_in[i] + 0.5)
+
+            # Check the bilinear interpolation matches the mean value of those 4 points (equivalent as its the middle)
+            assert raster_points_in[i] == np.mean([arr[xlow, ylow], arr[xupp, ylow], arr[xupp, yupp], arr[xlow, yupp]])
+
+        # Check bilinear extrapolation for points at 1 spacing outside from the input grid
+        points_out = (
+            [(-1, i) for i in np.arange(1, 4)]
+            + [(i, -1) for i in np.arange(1, 4)]
+            + [(4, i) for i in np.arange(1, 4)]
+            + [(i, 4) for i in np.arange(4, 1)]
+        )
+        points_out_xy = list(zip(*points_out))
+        raster_points_out = raster.interp_points(points_out_xy)
+        assert all(~np.isfinite(raster_points_out))
+
+        # To use cubic or quintic, we need a larger grid (minimum 6x6, but let's aim bigger with 50x50)
+        arr = np.flipud(np.arange(1, 2501).reshape((50, 50)))
+        transform = rio.transform.from_bounds(0, 0, 50, 50, 50, 50)
+        raster = gu.Raster.from_array(data=arr, transform=transform, crs=None, nodata=-9999)
+        raster.set_area_or_point(tag_aop, shift_area_or_point=False)
+
+        # For this, get random points
+        rng = np.random.default_rng(42)
+        index_x_in_rand = rng.integers(low=8, high=42, size=(10,)) + rng.normal(scale=0.3)
+        index_y_in_rand = rng.integers(low=8, high=42, size=(10,)) + rng.normal(scale=0.3)
+        points_x_rand, points_y_rand = raster.ij2xy(i=index_x_in_rand, j=index_y_in_rand, shift_area_or_point=shift_aop)
+
+        for method in ["nearest", "linear"]:
+            raster_points_mapcoords = raster.interp_points(
+                (points_x_rand, points_y_rand),
+                method=method,
+                force_scipy_function="map_coordinates",
+                shift_area_or_point=shift_aop,
+            )
+            raster_points_interpn = raster.interp_points(
+                (points_x_rand, points_y_rand),
+                method=method,
+                force_scipy_function="interpn",
+                shift_area_or_point=shift_aop,
+            )
+
+            # Not exactly equal in floating point precision since changes in Scipy 1.13.0,
+            # see https://github.com/GlacioHack/geoutils/issues/533
+            assert np.allclose(raster_points_mapcoords, raster_points_interpn)
+
+        # Check that, outside the edge, the interpolation fails and returns a NaN
+        index_x_edge_rand = [-0.5, -0.5, -0.5, 25, 25, 49.5, 49.5, 49.5]
+        index_y_edge_rand = [-0.5, 25, 49.5, -0.5, 49.5, -0.5, 25, 49.5]
+
+        points_x_rand, points_y_rand = raster.ij2xy(
+            i=index_x_edge_rand, j=index_y_edge_rand, shift_area_or_point=shift_aop
+        )
+
+        # Nearest doesn't apply, just linear and above
```

Comment:
linear is not in the list, as stated in the comment

Reply:
Also I do not understand why the test is not suited for nearest and linear? Because it is only half a pixel?

Reply:
Good catch, this was a relic of the previous implementation/tests; now it works for all methods. I have changed to use all :)